### PR TITLE
Implements domain tracking API.

### PR DESCRIPTION
**Pull request by BlakeHancock**
Implements the following API endpoints.

GET /domains/<domain>/tracking
PUT /domains/<domain>/tracking/open
PUT /domains/<domain>/tracking/click
PUT /domains/<domain>/tracking/unsubscribe

### CODE REVIEW

**Comment by BlakeHancock** [comment]
It looks like the automated checks are failing on code that already exists in the master.

**Comment by cdave3** [comment]
Can someone merge this please? I need it for a project I'm working on.

File: src/Model/Domain/Tracking/UpdateClickTrackingResponse.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    private $message;
+
+    public static function create(array $data): self
+    {
+        $model = new self();
+        if (isset($data['click'])) {
+            $model->click = ClickTrackingItem::create($data['click']);
+        }
+
+        $model->message = $data['message'] ?? null;
```
Comment (DavidGarciaCat) [inline src/Model/Domain/Tracking/UpdateClickTrackingResponse.php:31]: I cannot see any reference to `message` next to `click` when checking the documentation.
Can you please tell me where I can find it?

Thanks,

**Review by DavidGarciaCat — COMMENTED** [review]
Thanks for the PR.

I have just one question in regards to the `message` fields that the API returns (apparently) but I am unable to see it when checking the documentation. Maybe I'm just too tired.

Would you mind please telling me where I can find it?

Thanks,

File: src/Model/Domain/Tracking/UpdateClickTrackingResponse.php (hunk 1 above)
Comment (BlakeHancock) [reply src/Model/Domain/Tracking/UpdateClickTrackingResponse.php]: It appears to be vestigial remnants from referencing the way other calls of a similar nature were set up. I have removed them.

Comment (DavidGarciaCat) [reply src/Model/Domain/Tracking/UpdateClickTrackingResponse.php]: Thanks for the update.

It looks like now the PHPUnit tests are failing (please check Travis CI jobs) due to there's a discrepancy between the calls that we provide and the expected content as part of the tests.

If you can fix it, we can proceed.

Thanks,

**Comment by DavidGarciaCat** [comment]
This should be now implemented through https://github.com/mailgun/mailgun-php/pull/711

Closing this PR

**Pull request closed without merging**